### PR TITLE
[libx11] Optimize Build

### PR DESCRIPTION
**Pull request by Thomas1664**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

To beat:
- x64-windows: 28 min
- x86-windows: 28 min
- x64-windows-static-md: 26 min
- x64-windows-static: 26 min
- arm64-windows: 26 min

### CODE REVIEW

**Comment by Thomas1664** [comment]
> To beat:
> 
> * x64-windows: 28 min
> * x86-windows: 28 min
> * x64-windows-static-md: 26 min
> * x64-windows-static: 26 min
> * arm64-windows: 26 min

I can confirm that the build of libx11 is indeed significantly faster compared to the CI workflow. However, during PR workflows, the builds here were slower. I think the reason for that is that here all ports that depend on libx11 are also included which causes additional x11 ports to be present. Maybe there is an installation order issue hiding somewhere?

File: ports/libx11/optimize-configure.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/Makefile.am b/Makefile.am
+index b14f9d78..2df82fe3 100644
+--- a/Makefile.am
++++ b/Makefile.am
+@@ -4,7 +4,7 @@ else
+ ORDER=modules src
+ endif
+ # Order: nls before specs
+-SUBDIRS=include $(ORDER) nls man specs
++SUBDIRS=include $(ORDER) nls
```
Comment (MonicaLiu0311) [inline ports/libx11/optimize-configure.patch:10]: Does the upstream agree with this modification?

**Comment by MonicaLiu0311** [comment]
> > To beat:
> > 
> > * x64-windows: 28 min
> > * x86-windows: 28 min
> > * x64-windows-static-md: 26 min
> > * x64-windows-static: 26 min
> > * arm64-windows: 26 min
> 
> I can confirm that the build of libx11 is indeed significantly faster compared to the CI workflow. However, during PR workflows, the builds in this PR are slower. I think the reason for that is that here all ports that depend on libx11 are also built which causes additional x11 ports to be present. Maybe there is an installation order issue hiding somewhere?

[Installation order](https://dev.azure.com/vcpkg/public/_build/results?buildId=102161&view=logs&j=7922e5c4-0103-5f8f-ad17-45ce9bb98e80&t=66eef212-08d6-5e6c-a560-7d01149d0188&l=7372): Install indirect and direct dependencies of libx11 first, then install ports that have libx11 as a dependency.

File: ports/libx11/optimize-configure.patch (hunk 1 above)
Comment (Thomas1664) [reply ports/libx11/optimize-configure.patch:10]: Why would they need to? This is a vcpkg specific patch.

**Comment by MonicaLiu0311** [comment]
Record: Differences in the share folder after modification.
image

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**